benchopt/base.py

```diff
@@ -125,12 +126,18 @@ def install(cls, env_name=None, force=False):
         if force or not cls.is_installed(env_name=env_name):
             print(f"Installing {cls.name} in {env_name}:...",
                   end='', flush=True)
-            if cls.install_cmd == 'pip':
-                pip_install_in_env(*cls.requirements_install,
-                                   env_name=env_name)
-            elif cls.install_cmd == 'bash':
-                bash_install_in_env(cls.install_script, env_name=env_name)
-            print(" done")
+            try:
+                if cls.install_cmd == 'pip':
+                    pip_install_in_env(*cls.requirements_install,
+                                       env_name=env_name)
+                elif cls.install_cmd == 'bash':
+                    bash_install_in_env(cls.install_script, env_name=env_name)
+                print(" done")
+            except Exception as exception:
+                if PRINT_INSTALL_ERROR:
+                    raise exception
```

Comment:
I am not totally sure for this `raise`, my feeling with `print_install_error` was to print them but to not fail the benchmark. In particular, if one call `benchopt run <benchmark>`, this will fail the whole benchmark while the solver was not specifically requested.
But it is true that this is an option that you need to activate so I think it would make sense.

Maybe I would also change `PRINT_INSTALL_ERROR` to `RAISE_INSTALL_ERROR`?

Reply:
Oh yes I forgot to mention this.

When I run the benchmark with `PRINT_INSTALL_ERROR=True`, it definitely raises an error and stops the benchmark. With this behavior, I would indeed rename it `RAISE_INSTALL_ERROR`. If we prefer to just print the traceback and not raise an error, master also has to be fixed.

Reply:
So I would go for `RAISE_INSTALL_ERROR`! :smile: